src/pushsource/_impl/model/azure.py

```diff
@@ -0,0 +1,55 @@
+import re
+
+from frozenlist2 import frozenlist
+
+from .. import compat_attr as attr
+from .conv import in_, instance_of, optional_str, optional
+from .vms import VMIPushItem
+
+
+VHDGeneration = ["V1", "V2"]
+
+
+@attr.s()
+class VHDPushItem(VMIPushItem):
+    """A :class:`~pushsource.PushItem` representing a Virtual Hard Disk (or VHD).
+
+    For push items of this type, the :meth:`~pushsource.PushItem.src` attribute
+    refers to a file containing the VHD file..
+
+    This library does not verify that the referenced file is a valid
+    snapshot.
+    """
+
+    generation = attr.ib(type=str, default="V2", validator=optional(in_(VHDGeneration)))
+    """VHD generation. It can be ``V1`` or ``V2``."""
+
+    sku_id = attr.ib(type=str, default=None, validator=optional_str)
+    """SKU ID for the VM image generation, generally being the Plan ID."""
+
+    support_legacy = attr.ib(type=bool, default=False, validator=instance_of(bool))
+    """If the image ``generation == V2`` but also support ``V1``."""
+
+    legacy_sku_id = attr.ib(
+        type=str, default=None, validator=optional(in_(VHDGeneration))
+    )
+    """If the ``V2`` image also support ``V1`` what is the ``V1`` SKU ID."""
+
+    disk_version = attr.ib(type=str, default=None)
+    """The image's disk version according to Azure's format {int}.{int}.{int}"""
+
+    recommended_sizes = attr.ib(
+        type=list, default=attr.Factory(frozenlist), converter=frozenlist
+    )
+    """List of recommended VM sizes in Azure, for example ['a1-standard', 'a5', 'a7']"""
+
+    sas_uri = attr.ib(type=str, default=None, validator=optional_str)
+    """The SAS URI to be associated with the disk_version."""
+
+    @disk_version.validator
+    def _validate_regex(self, attribute, value):
```

Comment:
Since there is a class hierarchy here, I think this should be named with a double leading underscore `(def __validate_whatever)` to activate name mangling.  Otherwise, if someone ever introduced a method of the same name in a subclass or superclass of this, then one method is going to shadow another.

Also, though less important, `validate_regex` isn't a great name seeing as there could later be more than one attribute on the class which needs to be validated by a regex.  Naming it after the attribute being validated seems to make more sense.